### ISSUE TITLE
`num_classes` should not be required when using micro averaging

### ISSUE THREAD

**Issue by edmcman** [Lightning-AI/torchmetrics#2837]
## 🐛 Bug

When using micro averaging on multiclass_accuracy, it is required to pass in `num_classes`, even though it's not part of the calculation.

### To Reproduce

```py
from torchmetrics.functional.classification import multiclass_accuracy
import torch
x = torch.Tensor([0, 0, 0])
y = torch.Tensor([0, 1, 1])

print(multiclass_accuracy(x, y, average="macro", num_classes=2))
print(multiclass_accuracy(x, y, num_classes=2))
print(multiclass_accuracy(x, y, average="micro", num_classes=2))
try:
  print(multiclass_accuracy(x, y, average="micro"))
except:
  print("Need num_classes even for micro, why?")
```

### Expected behavior

In general, I expect the metric to be computed.  In the above code sample, I expect 0.33 to be returned, rather than an exception.

### Environment

- TorchMetrics version: 1.6.0
- Python version: 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
- PyTorch version: 2.5.0+cu121
- Colab https://colab.research.google.com/drive/1tVJJkktvEmRJgKGgzKrAsKYSx2c_hjSs?usp=sharing

**Comment by Borda** [comment]
> The micro-averaged accuracy metric is calculated by considering all the individual predictions across classes as a single pool, so it doesn’t require knowing the number of classes in advance.
